### ISSUE TITLE
Rename and single library

### ISSUE THREAD

**Issue by kegl** [paris-saclay-cds/ramp-workflow#19]
We should rename ramp-workflow to ramp-sdk. Existing starting kits will need to be updated.

We should have a single repo ramp and subrepos ramp-board, ramp-backend, and ramp-sdk. Then let users pip install those separately. Testing and PRs will be easier.

**Comment by agramfort** [comment]
+1 for sdk

**Comment by glemaitre** [comment]
Would it make sense to have submodule? Like in xgboost where you have a different repository for the `core` library, etc.

**Comment by rth** [comment]
> Would it make sense to have submodule? Like in xgboost where you have a different repository for the core library, etc.

It might be easier to have  a single repository with all the components or multiple repositories with fixed versions for each component IMO. Generally using git submodules makes things more complicated, particularly for new contributors ( e.g. can't just clone, also need to run `git submodule update --init --recursive`) etc.). I'm not sure of xgboost experience of it, but jupyter is not using them as far as I know, and here there are a few fairly negatives blog posts about git submodules in [[1](https://codingkilledthecat.wordpress.com/2012/04/28/why-your-company-shouldnt-use-git-submodules/)], [[2](http://slopjong.de/2013/06/04/git-why-submodules-are-evil/)], etc.

**Comment by glemaitre** [comment]
You are right. Speaking with joris, the issue tracker and PRs handling would be rather complicated.                                                                                                                                                                                                                                                                                                                                         Guillaume Lemaitre INRIA Saclay Ile-de-France / Equipe PARIETALguillaume.lemaitre@inria.fr - https://glemaitre.github.io/
